core/store.go

```diff
@@ -0,0 +1,222 @@
+package core
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/docker/libkv"
+	"github.com/docker/libkv/store"
+	"github.com/docker/libkv/store/boltdb"
+	"github.com/docker/libkv/store/consul"
+	"github.com/docker/libkv/store/etcd"
+	"github.com/docker/libkv/store/zookeeper"
+	log "github.com/Sirupsen/logrus"
+	"encoding/json"
+)
+
+func init() {
+	consul.Register()
+	etcd.Register()
+	zookeeper.Register()
+	boltdb.Register()
+}
+
+type Store struct {
+	ctx              *Context
+	kvstore          store.Store
+	storeServicePath string
+	storeBackendPath string
+	stopCh           chan struct{}
+}
+
+func NewStore(storeUrl, storeServicePath, storeBackendPath string, syncTime int64, context *Context) (*Store, error) {
+	uri, err := url.Parse(storeUrl)
+	if err != nil {
+		return nil, err
+	}
+	var backend store.Backend
+	if strings.EqualFold(uri.Scheme, "consul") {
```

Comment:
Looks like a `switch` statement would look better here instead of multiple `else if`-s.